README.md

```diff
@@ -1,39 +1,362 @@
+    implements ReviewSettingsSource<ReviewSettings> {
+  @override
+  Future<ReviewSettings> read() {
+    // TODO: Return stored review settings.
+    return Future.value(ReviewSettings());
+  }
+
+  @override
+  Future<void> write(ReviewSettings reviewSettings) {
+    // TODO: Update stored review settings.
+    return Future.value();
+  }
+}
+```
+
+### Using Dependency Injection
+
+Here is a simple code that does dependency injection using `get_it`.
+
+```dart
+   GetIt getIt = GetIt.instance;
+
+   // Register the ReviewPrompter implementation with GetIt
+   getIt.registerSingleton<ReviewPrompter>(ReviewPrompter());
+```
+
+### Adding your CustomReviewSettings
+
+First let's declare a new ReviewSettings named CustomReviewSettings with a favoriteJokesCount to track how many jokes were favorited in an hypothetical dad jokes application
+
+```dart 
+  class CustomReviewSettings extends ReviewSettings {
+    final int favoriteJokesCount;
+
+    const CustomReviewSettings({
+      this.favoriteJokesCount = 0,
+      super.primaryActionCompletedCount = 0,
+      super.secondaryActionCompletedCount = 0,
+      super.applicationLaunchCount = 0,
+      super.firstApplicationLaunch,
+      super.requestCount = 0,
+      super.lastRequest,
+    });
+
+    @override
+    CustomReviewSettings copyWith({
+      int? primaryActionCompletedCount,
+      int? secondaryActionCompletedCount,
+      int? applicationLaunchCount,
+      DateTime? firstApplicationLaunch,
+      int? requestCount,
+      DateTime? lastRequest,
+    }) {
+      return CustomReviewSettings(
+        primaryActionCompletedCount:
+            primaryActionCompletedCount ?? this.primaryActionCompletedCount,
+        secondaryActionCompletedCount:
+            secondaryActionCompletedCount ?? this.secondaryActionCompletedCount,
+        applicationLaunchCount:
+            applicationLaunchCount ?? this.applicationLaunchCount,
+        firstApplicationLaunch:
+            firstApplicationLaunch ?? this.firstApplicationLaunch,
+        requestCount: requestCount ?? this.requestCount,
+        lastRequest: lastRequest ?? this.lastRequest,
+        favoriteJokesCount: favoriteJokesCount,
+      );
+    }
+
+    CustomReviewSettings copyWithFavorite(int favoriteCount) {
+      return CustomReviewSettings(
+        primaryActionCompletedCount: super.primaryActionCompletedCount,
+        secondaryActionCompletedCount: super.secondaryActionCompletedCount,
+        applicationLaunchCount: super.applicationLaunchCount,
+        firstApplicationLaunch: super.firstApplicationLaunch,
+        requestCount: super.requestCount,
+        lastRequest: super.lastRequest,
+        favoriteJokesCount: favoriteCount,
+      );
+    }
+  }
+```
+
+Notes : 
+  1. It's important you define a copyWith that overrides the one from the ReviewSettings class while returning the newly added property. This ensures that the calls made to the copyWith method of ReviewSettings will be overriden by your implementation
```

Comment:
use `code` formating when you reference code elements (such as `copyWith`, `ReviewSettings`, etc.)

Reply:
Done.